helix-term/src/ui/text_decorations/diagnostics.rs

```diff
@@ -102,10 +102,11 @@ impl Renderer<'_, '_> {
             return 0;
         }
         let col = (col - self.renderer.offset.col) as u16;
+        let message = diag.message.replace('\n', "  ");
```

Comment:
this allocates a lot (once for each diagnostic during each frame). I intentionally didn't do something like this in the original PR for this reason. We will need to refactor `set_string_truncated` but since there are other consumers that is a larger change and I am worried about breaking other places that use that API (but maybe after this change we can update unicode-width?)